### PR TITLE
ModuleName <> ModulePath

### PR DESCRIPTION
**Pull request by HoshinoTented**
* ModuleName: a local name used in a local scope, such as a module context.
* ModulePath: an absolute path to a module.

### CODE REVIEW

File: base/src/main/java/org/aya/core/term/MetaTerm.java [hunk 1]
```diff
@@ -19,6 +19,10 @@ public record MetaTerm(
   @NotNull ImmutableSeq<@NotNull Arg<@NotNull Term>> contextArgs,
   @NotNull ImmutableSeq<@NotNull Arg<@NotNull Term>> args
 ) implements Callable {
+  public MetaTerm {
+
+  }
+
```
Comment (ice1000) [inline base/src/main/java/org/aya/core/term/MetaTerm.java:25]: What are these

File: base/src/main/java/org/aya/resolve/context/ModuleName.java [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+    @Override public @NotNull Qualified resolve(@NotNull String name) {
+      return new Qualified(ids.appended(name));
+    }
+
+    @Override public @NotNull Qualified concat(@NotNull ModuleName path) {
+      return new Qualified(ids.concat(path.ids()));
+    }
+
+    @Override public @NotNull String toString() {
+      return ids.joinToString(Constants.SCOPE_SEPARATOR);
```
Comment (ice1000) [inline base/src/main/java/org/aya/resolve/context/ModuleName.java:64]: QualifiedID.join

**Review by ice1000 — APPROVED** [review]
With minor nitpicking

File: base/src/main/java/org/aya/core/term/MetaTerm.java (hunk 1 above)
Comment (HoshinoTented) [reply base/src/main/java/org/aya/core/term/MetaTerm.java:25]: 😱 IDK!!

**Pull request merged**